lib/src/Base/Stat/SampleImplementation.cxx

```diff
@@ -1986,44 +1876,13 @@ SampleImplementation & SampleImplementation::operator += (const Point & translat
   return *this;
 }
 
-struct PositiveTranslationSamplePolicy
-{
-  const SampleImplementation & translation_;
-  SampleImplementation & output_;
-
-  PositiveTranslationSamplePolicy( const SampleImplementation & translation,
-                                   SampleImplementation & output)
-    : translation_(translation), output_(output) {}
-
-  inline void operator()( const TBB::BlockedRange<UnsignedInteger> & r ) const
-  {
-    for (UnsignedInteger i = r.begin(); i != r.end(); ++i) output_[i] += translation_[i];
-  }
-
-}; /* end struct TranslationSamplePolicy */
-
-struct NegativeTranslationSamplePolicy
-{
-  const SampleImplementation & translation_;
-  SampleImplementation & output_;
-
-  NegativeTranslationSamplePolicy( const SampleImplementation & translation,
-                                   SampleImplementation & output)
-    : translation_(translation), output_(output) {}
-
-  inline void operator()( const TBB::BlockedRange<UnsignedInteger> & r ) const
-  {
-    for (UnsignedInteger i = r.begin(); i != r.end(); ++i) output_[i] -= translation_[i];
-  }
-
-}; /* end struct TranslationSamplePolicy */
-
 SampleImplementation & SampleImplementation::operator += (const SampleImplementation & translation)
 {
   if (translation.getDimension() != dimension_) throw InvalidArgumentException(HERE) << "Error: the dimension of the given translation=" << translation.getDimension() << " does not match the dimension of the sample=" << dimension_;
   if (translation.getSize() != size_) throw InvalidArgumentException(HERE) << "Error: the size of the given translation=" << translation.getSize() << " does not match the size of the sample=" << size_;
-  const PositiveTranslationSamplePolicy policy( translation, *this );
-  TBB::ParallelFor( 0, size_, policy );
+  for(UnsignedInteger i = 0; i < size_; ++i)
+    for (UnsignedInteger j = 0; j < dimension_; ++j)
+      operator()(i, j) += translation(i, j);
   return *this;
```

Comment:
@dbarbier Would it be more efficient to use
```
for(UnsignedInteger index = 0; index < data_.getSize(); ++index)
    data_[index] += translation.data_[index];
```
as it avoids the computation of two linear indices from the exact same 2d indices?``

Reply:
Sure, it cannot hurt, I'll do that.

Reply:
@dbarbier I made some benchmarks on different implementations. It confirms your opinion: TBBs are useless here, your sequential approach gives a mean improvement of 33%. Switching to the 1D representation, the gain is now 36.4%, but using daxpy gives a gain of only 
27%. OpenBLAS is not the beast I imagined... Not bad, but not a tiger!

```
from openturns import *
from time import time

dimension = 10
size = 1000000
sample1 = Normal(dimension).getSample(size)
sample2 = Normal(dimension).getSample(size)
t0 = time()
sample3 = sample1 + sample2
t1 = time()
print("speed=", (dimension * size) / (t1 - t0), "adds/s")
# with TBBs
# speed= 111137419.9121352 adds/s
# speed= 111051791.28859988 adds/s
# speed= 110831998.90074465 adds/s
# speed= 96532181.96631515 adds/s
# speed= 110651302.1822633 adds/s
# wo TBBs
# speed= 142640113.17880073 adds/s
# speed= 141417103.0139148 adds/s
# speed= 148345435.18934423 adds/s
# speed= 139624897.63580316 adds/s
# speed= 147161848.75777596 adds/s
# wo TBBs linear
# speed= 146843444.8641779 adds/s
# speed= 148033910.50844586 adds/s
# speed= 147129843.02376568 adds/s
# speed= 147138617.4041774 adds/s
# speed= 148062129.34199378 adds/s
# with BLAS
# speed= 135709885.33119354 adds/s
# speed= 137525829.30851883 adds/s
# speed= 137667115.2394394 adds/s
# speed= 138009568.49635094 adds/s
# speed= 137355139.1463247 adds/s
```

Reply:
Hum, my figures are different, I have +10% gain with daxpy against woTBB.  Since MatrixImplementation uses daxpy, I will use it too here for consistency reasons.

BTW there is not much to gain with AXPY or GEMV, differences are noticeable with GEMM or factorizations on large matrices.